### PR TITLE
Add product to Dockerfile/docker-compose

### PR DESCRIPTION
**Pull request by adeebahmed**
Signed-off-by: AdeebAhmed <adeeb.ahmed@target.com>

### CODE REVIEW

File: docker-compose.yaml [hunk 1]
```diff
@@ -49,6 +49,17 @@ services:
       bash -c "
         /grid-pike-tp -vv -C tcp://validator:4004
       "
+
```
Comment (agunde406) [inline docker-compose.yaml]: Please rebase on master, this file is very different now that the we are using sabre

Comment (adeebahmed) [reply docker-compose.yaml]: Updated ✅

File: contracts/product/product.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: grid_product
+version: '1.0'
+wasm: target/wasm32-unknown-unknown/release/grid-product-tp.wasm
+inputs:
+  - '621dee01'
+  - '621dee0201'
+  - cad11d
```
Comment (agunde406) [inline contracts/product/product.yaml]: Missing single quotes around cad11d

File: contracts/product/product.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: grid_product
+version: '1.0'
+wasm: target/wasm32-unknown-unknown/release/grid-product-tp.wasm
+inputs:
+  - '621dee01'
+  - '621dee0201'
```
Comment (agunde406) [inline contracts/product/product.yaml]: This could also be  just 621dee02

File: docker-compose.yaml [hunk 4]
```diff
@@ -76,6 +76,24 @@ services:
...
+      dockerfile: contracts/product/Dockerfile
+    volumes:
+      - grid-shared:/grid-shared
+    entrypoint: |
+      bash -c "
+        while true; do curl -s http://grid-sawtooth-rest-api:8008/state | grep -q head; if [ $$? -eq 0 ]; then break; fi; sleep 0.5; done;
+        sabre cr --create product --key /grid-shared/my_key --owner $$(cat /grid-shared/my_key.pub) --url http://grid-sawtooth-rest-api:8008 --wait 30
+        sabre upload --filename product.yaml --key /grid-shared/my_key --url http://grid-sawtooth-rest-api:8008 --wait 30
+        sabre ns --create cad11d --key /grid-shared/my_key --owner $$(cat /grid-shared/my_key.pub) --url http://grid-sawtooth-rest-api:8008 --wait 30
+        sabre perm cad11d product --key /grid-shared/my_key --read --write --url http://grid-sawtooth-rest-api:8008 --wait 30
```
Comment (agunde406) [inline docker-compose.yaml]: Missing permissions for 621dee01 (read) and 621dee02 (read/write)

Comment (adeebahmed) [reply docker-compose.yaml]: ✅

File: contracts/product/product.yaml (hunk 3 above)
Comment (adeebahmed) [reply contracts/product/product.yaml]: ✅

File: contracts/product/product.yaml (hunk 2 above)
Comment (adeebahmed) [reply contracts/product/product.yaml]: ✅

File: contracts/product/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+# For Building Protobufs
+RUN curl -OLsS https://github.com/google/protobuf/releases/download/v3.5.1/protoc-3.5.1-linux-x86_64.zip \
+    && unzip -o protoc-3.5.1-linux-x86_64.zip -d /usr/local \
+    && rm protoc-3.5.1-linux-x86_64.zip
+
+COPY ./sdk /sdk
+
+RUN mkdir contracts \
+RUN USER=root cargo new --bin contracts/product
```
Comment (agunde406) [inline contracts/product/Dockerfile]: Combine these to lines to be 
```
RUN mkdir contracts \
   && USER=root cargo new --bin contracts/product
```

This does not build, please make sure that docker-compose up --build is successful

File: docker-compose.yaml [hunk 6]
```diff
@@ -76,6 +76,26 @@ services:
...
+        sabre perm 621dee01 product --key /grid-shared/my_key --read --url http://grid-sawtooth-rest-api:8008 --wait 30
+        sabre perm 621dee02 product --key /grid-shared/my_key --read --write --url http://grid-sawtooth-rest-api:8008 --wait 30
```
Comment (agunde406) [inline docker-compose.yaml]: The namespace registry for 621dee02 also needs to be created. Please make sure that these transaction are valid by checking the logs after `docker-compose up --build`

File: contracts/product/product.yaml [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+name: grid_product
+version: '1.0'
+wasm: target/wasm32-unknown-unknown/release/grid-product-tp.wasm
+inputs:
+  - '621dee01'
+  - '621dee02'
+  - 'cad11d'
+outputs:
+  - '621dee0201'
```
Comment (agunde406) [inline contracts/product/product.yaml]: This should also just be 621dee02

File: docker-compose.yaml (hunk 6 above)
Comment (adeebahmed) [reply docker-compose.yaml]: and for `621dee01` as well correct? or is that address accounted for in the namespace creations of `621dee01` Not too familiar with sabre namespace 😅

**Pull request merged**